XenModel/XenAPI-Extensions/SR.cs

```diff
@@ -120,7 +120,19 @@ public bool IsToolsSR()
 
         public string FriendlyTypeName()
         {
-            return GetFriendlyTypeName(GetSRType(false));
+            var srType = GetSRType(false);

```

Comment:
Pro of this solution: if the vendor changes the details of the SM plugin (e.g. type change between prototyping and productization) XC will pick it without changes.
Con: There is no localization. It would have to be provided by the plugin (but we don't have such implementation yet).